pkg/prometheus/resource_selector.go

```diff
@@ -704,6 +704,12 @@ func (rs *ResourceSelector) SelectScrapeConfigs(ctx context.Context, listFn List
 			rejectFn(sc, err)
 			continue
 		}
+		if sc.Spec.ProxyConfig != nil {
+			if err = sc.Spec.ProxyConfig.Validate(); err != nil {
+				rejectFn(sc, err)
+				continue
+			}
```

Comment:
we need something similar to https://github.com/prometheus-operator/prometheus-operator/blob/e2d552714ac0bb40f5b835e29ff0f7fe67cd8ecf/pkg/prometheus/resource_selector.go#L797-L801